File: LidarLitev3Sensor.java
package frc.robot.core;

import edu.wpi.first.wpilibj.TimedRobot;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
import edu.wpi.first.wpilibj.Counter;

public class LidarLitev3Sensor
{
    private Counter lidar;
    private double distance;

    public LidarLitev3Sensor(int lidarID)
    {
        lidar = new Counter(0); //plug the lidar into PWM 0
        lidar.setMaxPeriod(1.00); //set the max period that can be measured
        lidar.setSemiPeriodMode(true); //Set the counter to period measurement
        lidar.reset();
    }

    public void Update() 
    {
        if(lidar.get() < 1)
        {
            distance = 0;
        }
        else
        {
            distance = (lidar.getPeriod()*1000000.0/10.0); //convert to distance. sensor is high 10 us for every centimeter.
        }
        
        System.out.println("Distance: " + distance);
    }


    public double GetDistance()
    {
        return distance;
    }

}

File: LidarLitev3Sensor.cpp
#include "Core/LidarLitev3Sensor.h"
#include <iostream>

LidarLitev3Sensor::LidarLitev3Sensor(int lidarID)
    : lidar{lidarID}   // Initialize Counter with the PWM port
{
    lidar.SetMaxPeriod(units::time::second_t(1));         // Max measurable period
    lidar.SetSemiPeriodMode(true);  // Measure high-time of pulse
    lidar.Reset();
}

void LidarLitev3Sensor::Update() {
    if (lidar.Get() < 1) {
        distance = 0.0;
    } else {
        // Convert to cm: 10 microseconds per cm
        distance = (lidar.GetPeriod().value() * 1'000'000.0 / 10.0);
    }

    std::cout << "Distance: " << distance << std::endl;
}

double LidarLitev3Sensor::GetDistance() const {
    return distance;
}